lib/include/jxl/decode.h

```diff
@@ -913,6 +950,142 @@ JXL_EXPORT JxlDecoderStatus JxlDecoderSetJPEGBuffer(JxlDecoder* dec,
+ * not an error and returns 0.
+ *
+ * @param dec decoder object
+ * @return the amount of bytes the decoder has not yet written to of the data
+ * set by JxlDecoderSetBoxBuffer, or 0 if no buffer is set or
+ * JxlDecoderReleaseBoxBuffer was already called.
+ */
+JXL_EXPORT size_t JxlDecoderReleaseBoxBuffer(JxlDecoder* dec);
+
+/**
+ * Outputs the type of the current box, after a JXL_DEC_BOX event occured, as 4
+ * characters plus a null termination character. If the box is of type "brob",
+ * the underlying uncompressed box type is returned instead, not "brob".
+ *
+ * @param dec decoder object
+ * @param type buffer to copy the type into
+ * @return JXL_DEC_SUCCESS if the value is available, JXL_DEC_ERROR if not, for
+ *    example the JXL file does not use the container format.
+ */
+JXL_EXPORT JxlDecoderStatus JxlDecoderGetBoxType(JxlDecoder* dec, char type[5]);
+
+/**
+ * Outputs whether the given type equals the type returned by
+ * JxlDecoderGetBoxType.
+ *
+ * @param dec decoder object
+ * @param type box type to compare with
+ * @return JXL_TRUE if the box type is available and equal to the given type,
+ * JXL_FALSE otherwise.
+ */
+JXL_EXPORT JXL_BOOL JxlDecoderBoxTypeEquals(JxlDecoder* dec, const char* type);
+
+/**
+ * Outputs the type of the current box, after a JXL_DEC_BOX event occured, as 4
+ * characters plus a null termination character. If the box is of type "brob",
+ * the underlying uncompressed box type is returned instead, not "brob".
+ *
+ * @param dec decoder object
+ * @param type buffer to copy the type into
+ * @return JXL_DEC_SUCCESS if the value is available, JXL_DEC_ERROR if not, for
+ *    example the JXL file does not use the container format.
+ */
+JXL_EXPORT JxlDecoderStatus JxlDecoderGetBoxTypeRaw(JxlDecoder* dec,
+                                                    char type[5]);
+
+/**
+ * Outputs whether the given type equals the type returned by
+ * JxlDecoderGetBoxTypeRaw.
+ *
+ * @param dec decoder object
+ * @param type box type to compare with
+ * @return JXL_TRUE if the raw box type is available and equal to the given
+ * type, JXL_FALSE otherwise.
+ */
+JXL_EXPORT JXL_BOOL JxlDecoderBoxTypeRawEquals(JxlDecoder* dec,
+                                               const char* type);
+
+/**
+ * Outputs whether the raw box type equals "brob", indicating it is a brotli
+ * compressed box. Its underlying type is still returned with
+ * JxlDecoderGetBoxType. The data returned to the box buffer is transparently
+ * decompressed, unless JxlDecoderSetBoxModeRaw is enabled, so this function is
+ * not needed to just get the data, as long as the library is compiled with
+ * brotli.
+ *
+ * @param dec decoder object
+ * @param type box type to compare with
+ * @return JXL_TRUE if the box type is available and is "brob",
+ * JXL_FALSE otherwise.
+ */
+JXL_EXPORT JXL_BOOL JxlDecoderIsBoxCompressed(JxlDecoder* dec,
+                                              const char* type);
```

Comment:
I don't think we need the type here.

Also, isn't this function redundant wrt getting the Raw type?

Reply:
I'd say it isn't redundant, and I removed the wording of the comments here that make it say that it's the same as box type being "brob". Checking against "brob" is a bit too much like manually parsing, having `JxlDecoderIsBoxCompressed` lets the API parse the file format specifications details such as the string "brob" for you.

Reply:
Well, do we need the GetBoxTypeRaw function then? Why is that one useful if you have the IsCompressed one?